shell/components/nav/WindowManager/ContainerShell.vue

```diff
@@ -282,16 +288,32 @@ export default {
       this.socket.addEventListener(EVENT_DISCONNECTED, (e) => {
         this.isOpen = false;
         this.isOpening = false;
+
+        // If we had an error message, try connecting with the next command
+        if (this.errorMsg) {
+          if (this.commands.length) {
+            this.connect();
+          } else {
+            // Output last error with a new line and carriage return
+            this.terminal.write(`${ this.errorMsg }\n\r`);
```

Comment:
In the case that we're attempting to exec into a container and it is the initial OS we're trying, we would always attempt the incorrect OS command second, get an OS specific error back, over-write the actual error we'd gotten initially with an unhelpful OS-related error afterwards.

In order to rely on the error-first approach we'd need to parse the error and specifically be looking for the message `The system cannot find the file specified.: unknown` which I believe is Windows specific (provided different Windows releases don't use slightly different error messages) and a Linux machine would provide a different error when using the incorrect command such as `OCI runtime exec failed: exec failed: unable to start container process: exec: "cmd": executable file not found in $PATH: unknown` (once again, different base images might provide slightly different error messages that we'd need to test for).